.github/workflows/release.yml

```
@@ -40,10 +50,17 @@ jobs:
       - name: Pack release tarball
         id: pack
         run: |
-          TARBALL=$(npm pack | tail -n 1)
+          TARBALL=$(pnpm pack:release | tail -n 1)
           echo "tarball=${TARBALL}" >> "$GITHUB_OUTPUT"
+          echo "tarball_path=.release-artifacts/${TARBALL}" >> "$GITHUB_OUTPUT"
+
+      - name: Publish to npm
+        if: ${{ secrets.NPM_TOKEN != '' }}
```

actionlint (1.7.12)

[error] context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)